homeassistant/components/iotty/coordinator.py

```diff
@@ -0,0 +1,112 @@
+"""DataUpdateCoordinator for iotty."""
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+from datetime import timedelta
+import logging
+
+from iottycloud.device import Device
+from iottycloud.verbs import RESULT, STATUS
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import aiohttp_client, device_registry as dr
+from homeassistant.helpers.config_entry_oauth2_flow import OAuth2Session
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from . import api
+from .const import DOMAIN, LOGGER
+
+_LOGGER = logging.getLogger(__name__)
+
+UPDATE_INTERVAL = timedelta(seconds=30)
+
+
+@dataclass
+class IottyData:
+    """iotty data stored in the DataUpdateCoordinator."""
+
+    devices: list[Device]
+
+
+class IottyDataUpdateCoordinator(DataUpdateCoordinator[IottyData]):
+    """Class to manage fetching Iotty data."""
+
+    config_entry: ConfigEntry
+    _entities: dict[str, Entity]
+    _devices: list[Device]
+    _device_registry: dr.DeivceRegistry
+
+    def __init__(
+        self, hass: HomeAssistant, entry: ConfigEntry, session: OAuth2Session
+    ) -> None:
+        """Initialize the coordinator."""
+        _LOGGER.debug("Initializing iotty data update coordinator")
+
+        super().__init__(
+            hass,
+            LOGGER,
+            name=f"{DOMAIN}_coordinator",
+            update_interval=UPDATE_INTERVAL,
+        )
+
+        self.config_entry = entry
+        self._entities = {}
+        self._devices = []
+        self.iotty = api.IottyProxy(
+            hass, aiohttp_client.async_get_clientsession(hass), session
+        )
+        self._device_registry = dr.async_get(hass)
+
+    def set_entity(self, device_id: str, entity: Entity) -> None:
+        """Store iotty device within Hass entities."""
+        _LOGGER.debug("Storing device '%s' in entities", device_id)
+        self._entities[device_id] = entity
```

Comment:
Is this method still needed? It seems like it's only called from tests?

Reply:
removed!